src/CMakeLists.txt

```diff
@@ -30,3 +30,17 @@ if (KERNEL_HEADERS_DIR)
 MESSAGE(STATUS "Using KERNEL_HEADERS_DIR=${KERNEL_HEADERS_DIR}")
 target_compile_definitions(bpftrace PUBLIC KERNEL_HEADERS_DIR="${KERNEL_HEADERS_DIR}")
 endif()
+
+execute_process(
+  COMMAND git describe --abbrev=4 --dirty --always --tags
+  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+  OUTPUT_VARIABLE GIT_VERSION
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+
+add_definitions("-DVERSION=${GIT_VERSION}")
+
+configure_file(
```

Comment:
Since you used `add_definitions`, you don't need to use `configure_file` because the definiton will be set at compile time and will be available to all files you build.